### PR TITLE
fix Issue 21425 - Using va_start twice results in wrong values

### PR DESCRIPTION
**Pull request by WalterBright**
The problem was that va_start didn't actually do anything, as the initialization was done in the function prolog. This moves the initialization from the prolog to va_start.

I'm not entirely happy with it, so WIP.

### CODE REVIEW

**Comment by WalterBright** [comment]
Ready to go.

File: compiler/test/runnable/variadic.d [hunk 1]
```diff
@@ -1118,6 +1118,49 @@ void test15417()
 }
 
 
+/***************************************/
+// https://issues.dlang.org/show_bug.cgi?21425
```
Comment (PetarKirov) [inline compiler/test/runnable/variadic.d:1122]:
```suggestion
// https://issues.dlang.org/show_bug.cgi?id=21425
```

File: compiler/test/runnable/variadic.d [hunk 2]
```diff
@@ -1118,6 +1118,49 @@ void test15417()
...
+    va_end(ap);
+}
+
+void test21425()
+{
+    f5(0, 5);
+}
+
+/*********************************************/
+// https://issues.dlang.org/show_bug.cgi?23409
```
Comment (PetarKirov) [inline compiler/test/runnable/variadic.d:1148]:
```suggestion
// https://issues.dlang.org/show_bug.cgi?id=23409
```

File: compiler/test/runnable/variadic.d [hunk 3]
```diff
@@ -1118,6 +1118,49 @@ void test15417()
...
+
+void printf10(const(char)* fmt, ...){
+    for(int i = 0; i < 10; i++){
+        va_list args;
+        va_start(args, fmt);
+        vprintf(fmt, args);
```
Comment (PetarKirov) [inline compiler/test/runnable/variadic.d:1154]: The output of this function is not tested - can you add a [`RUN_OUTPUT`}(https://github.com/dlang/dmd/tree/master/compiler/test#test-configuration) section to the start of this file?

Comment (WalterBright) [reply compiler/test/runnable/variadic.d:1154]: It's not really necessary. The nature of the bug is it will crash.

Comment (WalterBright) [reply compiler/test/runnable/variadic.d]: replaced with vsprintf

**Review by PetarKirov — APPROVED** [review]

**Comment by WalterBright** [comment]
I'm glad to get this one fixed.

**Pull request merged**